pkg/services/sqlstore/live.go

```diff
@@ -0,0 +1,47 @@
+package sqlstore
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/models"
+)
+
+func (ss *SQLStore) SaveLiveChannelData(query *models.SaveLiveChannelDataQuery) error {
+	return inTransaction(func(sess *DBSession) error {
```

Comment:
this could be replaced by single upsert, but it seems to me that existing Dialect.UpsertSQL implemetation does not allow upserting with different columns for insert and update parts of query.